### PR TITLE
Don't suggest using the global EC when an implicit EC cannot be found.

### PR DESCRIPTION
**Pull request by DavidGregory084**
Fixes scala/bug#10808.

* I opened this against 2.12.x - is that okay or would you prefer to accept this in 2.13.x?
* Do you want a neg test that checks the error message is displayed?

### CODE REVIEW

**Comment by Ichoran** [comment]
Error messages should do as much as possible to help a programmer produce working code.  This change does exactly the opposite: makes it even harder to produce working code.  I appreciate that there are many cases where you need to care intimately about the execution context you are using, but it doesn't follow that the error message has to be unhelpful about where to find execution contexts and/or which one(s) might be a good first try.

**Comment by DavidGregory084** [comment]
@Ichoran let's take the discussion of this to the corresponding issue

File: src/library/scala/concurrent/ExecutionContext.scala [hunk 1]
```diff
@@ -54,8 +54,7 @@ import scala.annotation.implicitNotFound
  * Application callback execution can be configured separately.
  */
 @implicitNotFound("""Cannot find an implicit ExecutionContext. You might pass
-an (implicit ec: ExecutionContext) parameter to your method
-or import scala.concurrent.ExecutionContext.Implicits.global.""")
+an (implicit ec: ExecutionContext) parameter to your method.""")
```
Comment (He-Pin) [inline src/library/scala/concurrent/ExecutionContext.scala]: You might pass an `(implicit ec: ExecutionContext)` argument to your method or import one in the current scope; make sure choose the `ExecutionContext` wisely which matters a lot.

**Comment by SethTisue** [comment]
> I opened this against 2.12.x - is that okay or would you prefer to accept this in 2.13.x?

2.12.x is good

> Do you want a neg test that checks the error message is displayed?

that doesn't seem necessary to me.

File: src/library/scala/concurrent/ExecutionContext.scala [hunk 2]
```diff
@@ -54,8 +54,17 @@ import scala.annotation.implicitNotFound
  * Application callback execution can be configured separately.
  */
 @implicitNotFound("""Cannot find an implicit ExecutionContext. You might pass
-an (implicit ec: ExecutionContext) parameter to your method
-or import scala.concurrent.ExecutionContext.Implicits.global.""")
+an (implicit ec: ExecutionContext) parameter to your method.
+
+The ExecutionContext is used to configure how and on which
```
Comment (He-Pin) [inline src/library/scala/concurrent/ExecutionContext.scala:59]: how and where

File: src/library/scala/concurrent/ExecutionContext.scala [hunk 3]
```diff
@@ -54,8 +54,17 @@ import scala.annotation.implicitNotFound
...
+thread pools Futures will run, so the specific ExecutionContext
+that is selected is important.
+
+If your application does not define an ExecutionContext elsewhere,
+consider using Scala's global ExecutionContext by defining
+the following:
```
Comment (He-Pin) [inline src/library/scala/concurrent/ExecutionContext.scala:65]: You could use a specified ExecutionContext or do the following for simplicity.

File: src/library/scala/concurrent/ExecutionContext.scala [hunk 4]
```diff
@@ -54,8 +54,17 @@ import scala.annotation.implicitNotFound
  * Application callback execution can be configured separately.
  */
 @implicitNotFound("""Cannot find an implicit ExecutionContext. You might pass
-an (implicit ec: ExecutionContext) parameter to your method
-or import scala.concurrent.ExecutionContext.Implicits.global.""")
+an (implicit ec: ExecutionContext) parameter to your method.
```
Comment (viktorklang) [inline src/library/scala/concurrent/ExecutionContext.scala:57]: Perhaps something like this,

`an (implicit ec: ExecutionContext) parameter: to your method, to the constructor, or obtain it via other means.`

**Comment by adriaanm** [comment]
The new message is (quite!) enough of an improvement, that I'm going to merge this already. Further bikeshedding is welcome in followup PRs ;-)

**Comment by SethTisue** [comment]
thank you, David!

**Comment by DavidGregory084** [comment]
Ah sorry, I let this fall off my radar. Thanks!

**Pull request merged**